text/0000-min-rust-version.md

```diff
@@ -0,0 +1,179 @@
+- Feature Name: min_rust_version
+- Start Date: 2018-06-28
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Add `rust` field to the package section of `Cargo.toml` which will be used to
+specify crate's Minimum Supported Rust Version (MSRV):
+```toml
+[package]
+name = "foo"
+version = "0.1.0"
+rust = "1.30"
+```
+
+# Motivation
+[motivation]: #motivation
+
+Currently crates have no way to formally specify MSRV. As a result users can't
+check if crate can be built on their toolchain without building it. It also
+leads to the debate on how to handle crate version change on bumping MSRV,
+conservative approach is to consider such changes as breaking ones, which can
+hinder adoption of new features across ecosystem or result in version number
+inflation, which makes it harder to keep downstream crates up-to-date. More
+relaxed approach on another hand can result in broken crates for user of older
+compiler versions.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+`cargo init` will automatically create `Cargo.toml` with `rust` field equal to
+`rust="stable"` or `rust="nightly"` depending on the currently used toolcahin.
+On `cargo publish` cargo will take  currently used Rust compiler version and
+will insert it before uploading the crate. In other words localy your `Cargo.toml`
+willl still have `rust="stable"`, but version sent to crates.io will have
+`rust="1.30"` if you've used Rust 1.30. This version will be used to determine if
+crate can be used with the crate user's toolchain and to select appropriate
+dependency versions. In case if you have `rust="stable"`, but execute
+`cargo publish` with Nigthly toolcahin you will get an error.
+
+If you are sure that your crate supports older Rust versions (e.g. by using CI
+testing) you can change `rust` field accordingly. On `cargo publish` it will be
+checked that crate indeed can be built with the specified version. (though this
+check can be disabled with `--no-verify` option)
+
+For example, lets imagine that your crate depends on crate `foo` with 10 published
+versions from `0.1.0` to `0.1.10`, in versions from `0.1.0` to `0.1.5` `rust`
+field in the `Cargo.toml` sent to crates.io equals to "1.30" and for others to
+"1.40". Now if you'll build your project with Rust 1.33 `cargo` will select
+`foo v0.1.5`, and `foo v0.1.10` if you'll build your project with Rust 1.30 or
+later. But if you'll try to build your project with Rust 1.29 cargo will issue an
+error. Although this check can be disabled with `--no-rust-check` option.
+
+`rust` field should respect the following minimal requirements:
+- value should be equal to "stable", "nigthly" or to a version in semver format
```

Comment:
Shouldn't this allow a full string, like the date of a nightly build? For some tools which basically is built for a specific nightly (clippy comes to mind) this can be a useful tool. As well as that, it would be useful as AFAIK `1.xx.0-nighly` refers to all of the nightlys that have been built during the six weeks.

Reply:
I don't see much merit in using `1.xx.0-nightly` versions instead of the simple "nightly". I think it's a reasonable assumption that nightly toolchain will stay fairly up-to-date for most of the developers, and for crates which experience frequent breakage it's better to have finer-grained control provided by `nightly: 2018-01-01` approach.

Thanks for noticing the typo, I'll fix it!